src/main/java/easton/Easton.java

```diff
@@ -0,0 +1,396 @@
+        Action action;
+        String userInput;
+
+        while (!isFinished) {
+            userInput = ui.input();
+            Ui.divider();
+
+            try {
+                action = getActionFromInput(userInput);
+            } catch (IllegalActionException e) {
+                Ui.displayText(e.getMessage());
+                action = Action.INVALID;
+            }
+
+            switch (action) {
+            case BYE:
+                Ui.goodbye();
+                isFinished = true;
+                break;
+            case LIST:
+                Ui.displayText("Here are the tasks in your list:");
+                ui.list(tasks, (x) -> true);
+                break;
+            case MARK:
+                changeTaskStatus(userInput, true, "Nice! I've marked this task as done:");
+                saveTasks();
+                break;
+            case UNMARK:
+                changeTaskStatus(userInput, false, "OK, I've marked this task as not done yet:");
+                saveTasks();
+                break;
+            case TODO:
+                try {
+                    addTask(createToDo(userInput));
+                    saveTasks();
+                } catch (EmptyDescriptionException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            case DEADLINE:
+                try {
+                    addTask(createDeadline(userInput));
+                    saveTasks();
+                } catch (EmptyDescriptionException | InvalidFormatException | DateTimeFormatException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            case EVENT:
+                try {
+                    addTask(createEvent(userInput));
+                    saveTasks();
+                } catch (EmptyDescriptionException | InvalidFormatException | DateTimeFormatException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            case DELETE:
+                deleteTask(userInput);
+                saveTasks();
+                break;
+            case FIND:
+                try {
+                    findTasks(userInput);
+                } catch (EmptyDescriptionException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            default:
+                break;
+            }
+
+            Ui.divider();
+        }
+
+    }
+
+
+    /**
+     * Executes the program to start.
+     * @param args Environment arguments.
+     */
+    public static void main(String[] args) {
+        new Easton("task.csv").run();
+    }
+
+    /**
+     * Changes the status of a given task.
+     *
+     * @param input Input from the prompt.
+     * @param isDone Is the task done.
+     * @param message Message displayed to the user interface.
+     */
+    public void changeTaskStatus(String input, boolean isDone, String message) {
```

Comment:
I like how you use clear descriptive verbs for all of the methods in this class. Good job on following the naming convention!